frontend/src/components/homePage/HomePage.jsx

```diff
@@ -1,29 +1,50 @@
-import Carousel from "../eventCarousel/EventCarousel";
-import Header from "../header/Header";
-import SpinningBanner from "../spinningBanner/SpinningBaner";
+import EventsCarousel from '../carousel/EventsCarousel'
+import Header from '../header/Header'
+import SpinningBanner from '../spinningBanner/SpinningBaner'
+import { useState, useEffect } from 'react'
 
 function App() {
+    const MapsApiKey = import.meta.env.VITE_MAPS_API_KEY;
+    const url = import.meta.env.VITE_URL
+    let [events, setEvents] = useState()
+    let [ changeInEvents, setchangeInEvents] = useState(true)
+
+    useEffect(() => {
+        if (changeInEvents) fetchEvents();
+        setchangeInEvents(false)
+    }, [events]);
```

Comment:
Can you share a bit on what is the purpose of this state? I don't see changeInEvents to be updated anywhere else. Simply according to the code, if events state change, and changeInEvents flag is true (at the first of loading), and we fetch events again, and then changeInEvents flag is flipped to false and never be true again. So basically we are fetching events after the first time events state changes. What are we trying to achieve here?

Reply:
This one was in preparation of the "filter" feature

Reply:
We could take this conversation to your filter PR as well. I think Jerry made the same comment. Please help us understand why this flag and logic is needed for filter feature.